### PR TITLE
CODEOWNERS: Remove file

### PR DESCRIPTION
**Pull request by russellb**
I propose dropping the CODEOWNERS file.

The use of CODEOWNERS prevents the default notifications setting of "Participating and @mentioned" from working. If you are in the CODEOWNERS, you get added to every PR and will get all notifications accordingly. This is pretty painful on a busy repo. If a maintainer desires that level of email notifications, they can change their setting to "All Activity" for the repository.

CODEOWNERS is also not acting as access control to the repository. That access is managed via github teams, and those teams are granted different levels of access to the respository.

In summary, by not using a top-level entry like this, I feel we give people more control over their notifications while not losing any level of access controls on the repo. It allows people to stick to a workflow that best works for them.

If your PR is related to a contribution to the taxonomy, please, fill
out the following questionnaire. If not, replace this whole text and the
following questionnaire with whatever information is applicable to your PR.

### CODE REVIEW

**Comment by russellb** [comment]
Originally #718

Related to https://github.com/instructlab/instructlab/pull/938
